dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java

```diff
@@ -0,0 +1,294 @@
+
+                    // get contentlets only for main frame
+
+                    if (contentlets != null) {
+                        for (Contentlet contentlet : contentlets) {
+                            ctxMap.put("EDIT_CONTENT_PERMISSION" + contentlet.getIdentifier(),
+                                    permissionAPI.doesUserHavePermission(contentlet, PERMISSION_WRITE, user));
+                            ContentType type = contentlet.getContentType();
+                            if (type.baseType() == BaseContentType.WIDGET) {
+                                com.dotcms.contenttype.model.field.Field field = type.fieldMap()
+                                    .get("widgetPreexecute");
+                                if (field != null && UtilMethods.isSet(field.values())) {
+
+                                    String x = ctxMap.containsKey(WIDGET_PRE_EXECUTE) ? ""
+                                            : (String) ctxMap.get(WIDGET_PRE_EXECUTE);
+                                    ctxMap.put(WIDGET_PRE_EXECUTE, x + field.values() + "\n");
+                                }
+                            }
+
+                            // Check if we want to accrue the tags associated to each contentlet on
+                            // this page
+                            if (Config.getBooleanProperty("ACCRUE_TAGS_IN_CONTENTS_ON_PAGE", false)) {
+
+
+                                // Search for the tags associated to this contentlet inode
+                                List<Tag> contentletFoundTags = APILocator.getTagAPI()
+                                    .getTagsByInode(contentlet.getInode());
+                                if (contentletFoundTags != null) {
+                                    this.pageFoundTags.addAll(contentletFoundTags);
+                                }
+
+                            }
+
+                        }
+                    }
+                    // sets contentletlist with all the files to load per
+                    // container
+                    ctxMap.put("contentletList" + c.getIdentifier() + uniqueId, contentlets.stream()
+                        .map(con -> con.getIdentifier())
+                        .toArray(size -> new String[size]));
+                    ctxMap.put("totalSize" + c.getIdentifier() + uniqueId, new Integer(contentlets.size()));
+
+                }
+            }
+        }
+    }
+
+
+
+    public List<Tag> getPageFoundTags() {
+        return pageFoundTags;
+    }
+
+    public String getWidgetPreExecute() {
+        return (String) ctxMap.getOrDefault(WIDGET_PRE_EXECUTE, null);
+    }
+
+
+
+
+    public Context addAll(Context incoming) {
+
+        for (String key : this.ctxMap.keySet()) {
+            incoming.put(key, ctxMap.get(key));
+        }
+
+
+
+        return incoming;
+    }
+
+
+
+    public String asString() {
+
+
+        final StringWriter s = new StringWriter();
+        for (String key : this.ctxMap.keySet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)